packages/hydrogen/src/build.ts

```diff
@@ -0,0 +1,147 @@
+import { promises as fs } from 'fs';
+import { dirname, join, relative } from 'path';
+import {
+  debug,
+  download,
+  EdgeFunction,
+  execCommand,
+  getEnvForPackageManager,
+  getNodeVersion,
+  getSpawnOptions,
+  glob,
+  readConfigFile,
+  runNpmInstall,
+  runPackageJsonScript,
+  scanParentDirs,
+} from '@vercel/build-utils';
+import type { BuildV2, PackageJson } from '@vercel/build-utils';
+
+export const build: BuildV2 = async ({
+  entrypoint,
+  files,
+  workPath,
+  config,
+  meta = {},
+}) => {
+  const { installCommand, buildCommand } = config;
+
+  await download(files, workPath, meta);
+
+  const mountpoint = dirname(entrypoint);
+  const entrypointDir = join(workPath, mountpoint);
+
+  // Run "Install Command"
+  const nodeVersion = await getNodeVersion(
+    entrypointDir,
+    undefined,
+    config,
+    meta
+  );
+
+  const spawnOpts = getSpawnOptions(meta, nodeVersion);
+  const { cliType, lockfileVersion } = await scanParentDirs(entrypointDir);
+
+  spawnOpts.env = getEnvForPackageManager({
+    cliType,
+    lockfileVersion,
+    nodeVersion,
+    env: spawnOpts.env || {},
+  });
+
+  if (typeof installCommand === 'string') {
+    if (installCommand.trim()) {
+      console.log(`Running "install" command: \`${installCommand}\`...`);
+      await execCommand(installCommand, {
+        ...spawnOpts,
+        cwd: entrypointDir,
+      });
+    } else {
+      console.log(`Skipping "install" command...`);
+    }
+  } else {
+    await runNpmInstall(entrypointDir, [], spawnOpts, meta, nodeVersion);
+  }
+
+  // Copy the edge entrypoint file into `.vercel/cache`
+  const edgeEntryDir = join(workPath, '.vercel/cache/hydrogen');
```

Comment:
Seems like a fine place to put the file. I consider it a "build artifact" of `@vercel/hydrogen` builder. I'm open to suggestions though if you have a preferred location to copy the file to.

Reply:
What I mean is that I don't think it needs to be copied to the disk at all. Its similar to node-bridge where you can use a FileBlob to copy the contents into the build output without creating a file on disk.

Reply:
Not really. The file is referenced via env var when invoking `shopify hydrogen build`, so their bundler needs access ot the file. It also need to be "within" the project source code so that it has access to `node_modules`.